code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -45,17 +45,137 @@
 	return 1	//No drifting in space for space carp!	//original comments do not steal
 
 /mob/living/simple_animal/hostile/carp/FindTarget()
-	. = ..()
-	if(.)
-		custom_emote(1,"nashes at [.]")
+	if(!faction) //No faction, no reason to attack anybody.
+		return null
+
+	var/atom/T = null
+	var/lowest_health = INFINITY // Max you can get
+	stop_automated_movement = 0
+	for(var/mob/living/L in targets)
+		if((L.faction == src.faction) && !attack_same)
+			continue
+		if(L in friends)
+			continue
+
+		if(!L.stat && (L.health < lowest_health))
+			lowest_health = L.health
+			T = L
+	if(isnull(T))
+		for(var/atom/A in targets)

```

Comment:
See my comment about generalization. You could basically _almost_ remove this function override with that.